### ISSUE TITLE
[Feature Request] "Post-It Note" Object

### ISSUE THREAD

**Issue by ChaoShaoBaoo** [UTDNebula/planner#313]
# Overview

Planner currently assumes user knows degree(s) requirements and does NOT explicitly list out the requirements in a simple, easily refer-able manner for user during the construction of their 4yrplan.

# Proposal
![drawnexplanation](https://user-images.githubusercontent.com/115287771/202074825-928811c5-a71a-45fe-b547-a8f7116f6a5f.jpg)
^ Large Rectangle = Screen w/ Planner
Extension on bottom ("Fully expanded example, Communication, +") would be an expansion of yellow highlighted post-it note. The extension is NOT a pop-out of the screen, it is bounded by the post-it note.

Currently envisioning a referable "post-it note" type web object that includes expandable drop downs: 
1. "Core (X credits needed),"
2. "Major (X credits needed),"
3. "Elective (X credits needed),"
4. [For certain majors] "Concentration/Specialization (X credits needed),"

The headings of the drop downs would be colored red (as well as the text # of needed credits), and they (both the heading and text) would turn green once the user fulfills the credit requisites for that particular section in their plan.

Once expanded, the requirements would be elaborated upon in a similar format to the Catalog Reference.
- Basically copy paste of Catalog info: title + course number of fulfilling courses.
- However, if user already has courses to fulfill requisites in planner space, courses will be green vs. blue/black default.

Courses would also ideally be drag-able from post-it listings into planner space. (e.g. user could drag "COMM 1311" into semester like they would a "boxed course" from the search bar.)
- Only exception is if course is green and already listed in planner;
- Could either not allow to drag at all. OR Could allow to drag, but will then notify red marker similar to the "pre/co-req not yet met!" saying "repeated course!" and will highlight the first instance of the course in a yellow/green color while repeated course is being engaged with.

Defaulted to not show full post-it: simply say "Requirements (not) fulfilled!" including an (i) to expand the post-it, and if expanded would have exit-able "x" in upper right corner of post-it (*see drawnexplanation above)
![post-it note collapsed vs](https://user-images.githubusercontent.com/115287771/202075519-8c1dc33f-7dab-4ac8-9a0c-a3d1d5d052f3.png)

## Supplementary Information
For more established reference of something similar:
![image](https://user-images.githubusercontent.com/115287771/202068872-6a09d18b-7b6c-4551-9e4f-2ec7223cf911.png)

For the 4th drop down, I know that Neuroscience has specific tracks (Industry, Research, and Medical) that students are required to declare for their degree. It's basically a concentration. We'd need to make sure courses required were dependent upon chosen track and do NOT require courses from other tracks. (courses from other tracks should be visible and usable to fulfill general requirements, but should not be a requirement for tracks not associated with.)

Mathematics has something similar dubbed, "Specializations."

Also should be coordinated with Transfer Credit #175 .

**Comment by ZzRanger** [comment]
I just realized what if we have the post-it note be the place to search for courses and then the left sidebar be the place to list the user's unfulfilled degree requirements? Basically instead of "Requirements Not Fulfilled" in the bottom right, it will say "Search Courses", and then on the left sidebar it has a list of requirements the user has to satisfy for their degree.

Since there are a lot of degree requirements, I think it would make sense to present all that information in a bigger container. It also makes sense to make the search courses functionality a collapsible since the user won't always be using that to add courses. What do you think?

**Comment by ChaoShaoBaoo** [comment]
Hmm. I understand where you’re coming from and I agree that there’s a lot of information that we’d be squishing in the post-it note, but I’d still prefer having the left side and more space allocated to course searching. 

The main action that the user is doing is course searching, so the user should have the most space to do what they want to do (btw this is not researched fact, I’m just hypothesizing). If we were to switch them around, I feel like the user would feel rather squished and uncomfortable having to work with such a restricted (albeit expandable) space, and would be much more comfortable using larger allocated area for the personal/interactive searching, and then the smaller pop-up-like post-it object for less personalized actions like checking off requirements. 

Yes, it’s more text in less space, but the information in the text is not the fundamental thing the user is here to do: find personalized classes effectively and place them into an organizable plan. Having a reference is helpful, but not primary to their purpose being here. 

It’s definitely an interesting idea though and I’m always open to discussion. Thanks for sharing!

**Comment by thekevinge** [comment]
Hey @ChaoShaoBaoo @ZzRanger ,

So apparently one of the larger complaints with our tool was horizontal scrolling. Caleb suggested sometimes along the lines of the Dartmouth's degree planner (D-planner), so I'm working on the UI for that at the moment (see below).

<details>

<summary>Click to see D-planner UI here</summary>

<img src="https://user-images.githubusercontent.com/76929931/208001726-502d3179-7103-4a27-806b-a1a30060c29d.png" />

</details>


In theory, if we adopted D-planner's UI, we should be able to keep the course list on the left and present a fully expanded degree requirements on the right side. For mobile, we could shrink it to an expandable post-it.

So basically take the D-planner UI and replace the 4th column (they have trimesters) with our degree requirements sidebar.

Let me know your thoughts.

**Comment by ChaoShaoBaoo** [comment]
Ooo. Okay, so I like this a lot. It does what I wanted to do, but more professionally/permanently. 

Were we to adopt a D-planner-type UI, I would much rather have the degree requirements be on the left with the "search course list" area and NOT on the right. Reason being, we're already squishing the 4-year planner part of planner into this kind of box format to ideally take up only one screen's worth of space. I'd prefer to relieve more space for the planner by combining the deg reqs with the course search into one column vs. separating them (leaving less space for the actual planner). 

Also, I have reasonable doubt that we'd be able to fit everyone's classes legibly into one screen with boxes as shown (most full-time would take 4-6 classes, so more vertical space). Are we planning on making planner exclusively vertical scrolling? (e.g. two (or three, if including summer) boxes per row (each row indicates a year), and then having the user scroll through the years vertically?) 

Thanks for the update, Kevin!

**Comment by ZzRanger** [comment]
Wait this looks kinda fire uhhh closing issue for now but going to slap this somewhere else
